dotnet/src/SemanticKernel.Abstractions/KernelBuilder.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>Provides a builder for constructing instances of <see cref="Kernel"/>.</summary>
+internal sealed class KernelBuilder : IKernelBuilder
+{
+    /// <summary>Whether to allow a call to Build.</summary>
+    /// <remarks>As a minor aid to help avoid misuse, we try to prevent Build from being called on instances returned from AddKernel.</remarks>
+    internal bool _allowBuild;
```

Comment:
nit: Since it's not private and we use it outside, should we rename it from `_allowBuild` to `AllowBuild`? Convert it to property?